src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs

```diff
@@ -23,16 +22,11 @@ public ToolInstallCommand(
             ToolInstallLocalCommand toolInstallLocalCommand = null)
             : base(parseResult)
         {
-            _toolInstallLocalCommand =
-                toolInstallLocalCommand
-                ?? new ToolInstallLocalCommand(_parseResult);
+            _toolInstallLocalCommand = toolInstallLocalCommand;
 
-            _toolInstallGlobalOrToolPathCommand =
-                toolInstallGlobalOrToolPathCommand
-                ?? new ToolInstallGlobalOrToolPathCommand(_parseResult);
+            _toolInstallGlobalOrToolPathCommand = toolInstallGlobalOrToolPathCommand;
```

Comment:
I'm actually a bit confused why this was changed to no longer initialize the fields, but instead, changed to only `new` a command within `Execute` instead. I don't see the benefit in the change, but maybe I'm missing something... 🤔

Reply:
Previously, we were eagerly initializing the commands, but we only actually need one of them...why make two if you're only going to use one? Doing it in Execute means we know which we need, and we don't have to make the other.

Reply:
Ah, makes sense.